packages/gestalt/src/Masonry.js

```diff
@@ -401,6 +401,8 @@ export default class Masonry<T: { ... }> extends ReactComponent<Props<T>, State<
       isVisible = true;
     }
 
+    const isRtl = document?.dir === 'rtl';
```

Comment:
@dangerismycat major thing I would test for is SSR when we do the bump in pinboard since `document` is not available on the server.

Reply:
Given the optional chaining, I wouldn't think we'd see an error there. Would the main concern be layout thrash?

Reply:
@dangerismycat yes, when server rendering the items, `document` will be `undefined` which means `isRtl` will be `false` even if `html dir="rtl"` is set.

However, we also have these 3 server render style files in pinboard which are used during server side render. So those might need to be adjusted to make SSR working correctly without the layout shifting from LTR to RTL.

- `webapp/app/packages/gestaltExtensions/Masonry/styles/getFlexibleGridServerStyles.js`
- `webapp/app/packages/gestaltExtensions/Masonry/styles/getDuploServerSideStyles.js`
- `webapp/app/packages/gestaltExtensions/Masonry/styles/getClassicGridServerStyles.js`

# Notes

* It would be great if we could move those "ServerStyles" into Gestalt so we don't have this weird interconnection between the 2
* These "ServerStyle" files work by relying on the `static` class name we give the rendered items during SSR: https://github.com/pinterest/gestalt/blob/c8193d9b33816f07ffd11605793b75ed92aa9211/packages/gestalt/src/Masonry.js#L481 . That's a different render output than what we do on the client with `renderMasonryComponent`: https://github.com/pinterest/gestalt/blob/c8193d9b33816f07ffd11605793b75ed92aa9211/packages/gestalt/src/Masonry.js#L404

Reply:
Ah I wasn't aware of those files, thanks! I just added a ticket for myself to migrate those into Gestalt.

Reply:
fwiw - when we migrate those into Gestalt, we should consolidate as much as possible vs maintaining the multiple forks

Reply:
agreed!